mantatail.py

```diff
@@ -56,28 +89,46 @@ def run_server_forever(self) -> None:
 
 
 def close_socket_cleanly(sock: socket.socket) -> None:
+    """Ensures that the connection to a client is closed cleanly without errors."""
+    # The code is based on this blog post:
     # https://blog.netherlabs.nl/articles/2009/01/18/the-ultimate-so_linger-page-or-why-is-my-tcp-not-reliable
     try:
         sock.shutdown(socket.SHUT_WR)
         sock.settimeout(10)
         sock.recv(1)  # Wait for client to close the connection
     except OSError:
         # Possible causes:
-        #   - Client decided to keep its connection open for more than 10sec.
-        #   - Client was already disconnected.
-        #   - Probably something else too that I didn't think of...
+        # - Client decided to keep its connection open for more than 10sec.
+        # - Client was already disconnected.
+        # - Probably something else too that I didn't think of...
         pass
 
     sock.close()
 
 
 def recv_loop(state: ServerState, user_host: str, user_socket: socket.socket) -> None:
+    """
+    Instantiates UserConnection and listens for commands/messages from the client,
+    parses them and sends them to appropriate "handle_" function in "commands".
+
+    IRC Messages are formatted "bytes(COMMAND args\r\n)"
+    Note that netcat uses "\n" in place of "\r\n".
+
+    Ex: b"JOIN #foo\r\n"
+    Ex: b"PRIVMSG #foo :This is a message\r\n"
+
+    COMMAND is parsed to "handle_command", ex. "handle_join".
+    "getattr(commands, "handle_join")" checks if there is an appropriate handler function
+    in commands.py.
+    If there is, the handler function is called. Otherwise, an unknown command error is
+    sent to the client.
```

Comment:
This contains lots of technical details (getattr, error handling, the fact that the function name is saved to a string), and isn't easy to read (nested quotes, examples are a bit unclear). How about something like this:

```suggestion
    To handle a command FOO, a function named handle_foo() in commands.py is called.
    For example, "PRIVMSG #foo :this is a message\r\n" results in a call like this:

        commands.handle_privmsg(state, user, ["#foo", "this is a message"])
```